### DISCUSSION TITLE
Change color temp of bulbs with rotation?

### DISCUSSION THREAD

**Discussion by pdawg17** [cgiesche/streamdeck-homeassistant#209] [Help and Support] [answered]
I am trying to adjust the color temp of my Hue bulbs with rotation but cannot get it working correctly. Is it possible? I may be interpreting how this plugin works but if I use "Kelvin", rotationAbsolute is only a number between 0 and 255 and rotationPercent is 0-100%, does it not work because Kelvin is a range of numbers that do not fit for these categories?

**Comment by cgiesche** [comment, marked as answer]
With a bit of Math-Foo it works (for me with my hue lights):
Service Data for "Turn On" Service:
```
{% set min = 2200 %}
{% set max = 6535 %}
{
"kelvin": {{ min + rotationAbsolute * ((max - min) / 255) }}
}
```
`min` = Minimum color temp supported by bulb
`max` = Maximum color temp supported by bulb

You can validate all values and the outcome with custom labels:
```


{{min_color_temp_kelvin}}-{{max_color_temp_kelvin}}
{{color_temp_kelvin}}
```
![grafik](https://github.com/cgiesche/streamdeck-homeassistant/assets/5123180/19652680-35ce-46a4-8639-85364b80b580)

---
Bonus:
You can create nice "bright cold to dim warm" effects, too:
```
{% set min = 2200 %}
{% set max = 4000 %}
{
"kelvin": {{ min + rotationAbsolute * ((max - min) / 255) }},
"brightness": {{ rotationAbsolute }}
}
```

**Comment by pdawg17** [comment]
> With a bit of Math-Foo it works (for me with my hue lights): Service Data for "Turn On" Service:
> 
> ```
> {% set min = 2200 %}
> {% set max = 6535 %}
> {
> "kelvin": {{ min + rotationAbsolute * ((max - min) / 255) }}
> }
> ```
> 
> `min` = Minimum color temp supported by bulb `max` = Maximum color temp supported by bulb
> 
> You can validate all values and the outcome with custom labels:
> 
> ```
> 
> 
> {{min_color_temp_kelvin}}-{{max_color_temp_kelvin}}
> {{color_temp_kelvin}}
> ```
> 
> ![grafik](https://user-images.githubusercontent.com/5123180/284005180-19652680-35ce-46a4-8639-85364b80b580.png)
> 
> Bonus: You can create nice "bright cold to dim warm" effects, too:
> 
> ```
> {% set min = 2200 %}
> {% set max = 4000 %}
> {
> "kelvin": {{ min + rotationAbsolute * ((max - min) / 255) }},
> "brightness": {{ rotationAbsolute }}
> }
> ```

WOW! Thank you! Works great! That's way above my Math-Foo level...

**Comment by Samywamy10** [comment]
Is it possible to create something similar for colours? Requires a bit more maths. I mucked around with it but couldn't get anything reliable

**Reply by alexwhelan1997** [reply]
@cgiesche this would be cool :)

**Reply by cgiesche** [reply]
```
{
"hs_color": [{{(360 / 255 * rotationAbsolute) | round }},100]
}
```

Cycles through all colors. Saturation is always 100% in this example.
